redaxo/src/core/lib/util/editor.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+class rex_editor
+{
+    use rex_factory_trait;
+
+    // see https://github.com/filp/whoops/blob/master/docs/Open%20Files%20In%20An%20Editor.md
+    // keep this list in sync with $supportedEditors in system.settings.php
+    private $editors = [
+        'atom' => 'atom://core/open/file?filename=%file&line=%line',
+        'emacs' => 'emacs://open?url=file://%file&line=%line',
+        'idea' => 'idea://open?file=%file&line=%line',
+        'macvim' => 'mvim://open/?url=file://%file&line=%line',
+        'phpstorm' => 'phpstorm://open?file=%file&line=%line',
+        'sublime' => 'subl://open?url=file://%file&line=%line',
+        'textmate' => 'txmt://open?url=file://%file&line=%line',
+        'vscode' => 'vscode://file/%file:%line',
+    ];
+
+    public function urlFromFile($filePath, $line)
+    {
+        // make internal urls work with parse_url()
+        $filePath = str_replace('rex:///', 'rex://', $filePath);
+
+        $parsedUrl = parse_url($filePath);
+        // rex:// internal url mapping to backend form-edit urls
+        if ($parsedUrl['scheme'] === 'rex') {
+            if ($parsedUrl['host'] === 'template') {
+                $templateId = ltrim($parsedUrl['path'], '/');
+                return rex_url::backendPage('templates', ['function' => 'edit', 'template_id' => $templateId]);
+            }
+            if ($parsedUrl['host'] === 'module') {
+                $moduleId = (int) ltrim($parsedUrl['path'], '/');
+                return rex_url::backendPage('modules/modules', ['function' => 'edit', 'module_id' => $moduleId]);
+            }
+        }
+
+        $systemEditor = rex::getProperty('system_editor');
+        $editorUrl = $this->editors[$systemEditor];
+
+        $editorUrl = str_replace('%line', rawurlencode($line), $editorUrl);
+        $editorUrl = str_replace('%file', rawurlencode($filePath), $editorUrl);
```

Comment:
funktioniert bei mir unter Linux nur ohne `rawurlencode()`. Werde das mal auf Mac OS ausprobieren.

Reply:
Unter Mac OS ohne Probleme, lässt man dann den Linux support weg?

Reply:
Das hab ich 1:1 aus whoops übernommen... ist jemandem was bekannnt wg inkompatibilitäten dahingehend?

Reply:
ah, unter Linux wird `xdg-open` verwendet und der kann `%20` nicht wieder in ein Space umwandeln.
siehe [stackexchange.com](https://unix.stackexchange.com/questions/169241/xdg-open-doesnt-handle-percent-encoded-file-urls-with-lxde).

Reply:
da hat man quasi keine Möglichkeit gegenzuwirken. Und ein `PHP_OS`-Check möchte ich vermeiden. Also würde ich das erstmal so lassen, wie es aktuell ist.

Reply:
Geht es mit urlencode() übergreifend?

Reply:
leider nein

Reply:
Hab das mal als upstream issue erfasst
https://github.com/filp/whoops/issues/581

Reply:
Ich habe mal geschaut, wie es Symfony macht. Die nutzen kein urlencode, sondern schreiben die URL mit Leerzeichen ins HTML, wie im Quelltext auf dem Screenshot zu sehen.
In der Statuszeile sieht man, dass mein Browser da automatisch `%20` draus macht.
Die Slashes bleiben unkodiert.

Also würde das bei dir @bloep denn dann funktionieren, wenn wir auch gar nicht (raw)urlencode verwenden würden?
Wenn das hier so stimmt, dann ist es ok, URLs unkodiert ins HTML zu schreiben: https://stackoverflow.com/a/3705601/3837585 (htmlspecialchars braucht man weiterhin).

![screenshot 2018-06-24 14 52 54](https://user-images.githubusercontent.com/330436/41819296-52b10504-77be-11e8-9128-69aca29ee0ed.png)

Reply:
@gharlan Ohne `rawurlencode()` also mit unkodierter URL funktioniert es :+1: 

Als Beispiel:
`phpstorm://open?file=/var/www/html/redaxo/src/addons/project/boot.php&amp;line=1`

Reply:
Dann würde ich das rausnehmen.

Reply:
Ok